### PR TITLE
[APM-34] M1/ARM support (runtime)

### PR DESCRIPTION
**Pull request by KVS85**
That should give us release packages for bot native ARM and multiarch images for 3.10 with the following naming:
- Linux:
  - DEB: `<PACKAGE NAME AND VERSION>_(amd64 | arm64).deb`
  - RPM: `<PACKAGE NAME AND VERSION>.(x86_64 | aarch64).rpm`
  - TAR.GZ:  `<PACKAGE NAME AND VERSION>_(x86_64 | arm64).tar.gz`
- Mac:
  - DMG: `<PACKAGE NAME AND VERSION>.(x86_64 | arm64).dmg`
  - TAR.GZ: `<PACKAGE NAME AND VERSION>_(x86_64 | arm64).tar.gz`
- Windows:
  - EXE: `<PACKAGE NAME AND VERSION>_amd64.exe`
  - ZIP: `<PACKAGE NAME AND VERSION>_amd64.zip`
Docker:
  - Separate (3.10+): `arangodb/(arangodb | enterprise)[-(preview | test)]:<VERSION TAG>-(amd64 | arm64v8)`
  - Multiarch (3.10+): `arangodb/(arangodb | enterprise)[-(preview | test)]:<VERSION TAG>`

3.9- packages and images naming will remain as they are.

### CODE REVIEW

**Review by dothebart — APPROVED** [review]

**Comment by KVS85** [comment]
- [ ] Jenkins: https://jenkins01.arangodb.biz/view/3.devel/job/arangodb-devel-packages-nightly/729/

File: helper.fish [hunk 1]
```diff
@@ -750,6 +750,22 @@ function buildTarGzPackageHelper
...
+        set arch "_$ARCH"
+      case '*'
+        if string match --quiet --regex '^arm64$|^aarch64$' $ARCH >/dev/null
+          set arch "_arm64"
+        else
+          echo "fatal, unknown architecture $ARCH for TGZ"
+          exit 1
+        end
+    end
+  end
```
Comment (mpoeter) [inline helper.fish:767]: This code seems to be duplicated in several places. Can we move it into a function and reduce code duplication?

**Review by mpoeter — COMMENTED** [review]
LGTM, although I do not feel qualified enough to do a review for this code.

**Comment by KVS85** [comment]
- [ ] https://jenkins01.arangodb.biz/job/arangodb-devel-packages-nightly/730/

File: helper.fish (hunk 1 above)
Comment (KVS85) [reply helper.fish:767]: Yes we can do it, but not for this PR.

**Review by joerg84 — APPROVED** [review]

**Pull request merged**